test/log-iterator.spec.js

```diff
@@ -0,0 +1,243 @@
+      repo: config.defaultIpfsConfig.repo + '-log-join' + new Date().getTime()
+    })
+
+    before(async () => {
+      rmrf.sync(ipfsConfig.repo)
+      testIdentity = await IdentityProvider.createIdentity({ id: 'userA', identityKeysPath, signingKeysPath })
+      ipfs = await startIpfs(IPFS, ipfsConfig)
+    })
+
+    after(async () => {
+      await stopIpfs(ipfs)
+      rmrf.sync(ipfsConfig.repo)
+    })
+
+    describe('Basic iterator functionality', () => {
+      let log1
+
+      beforeEach(async () => {
+        log1 = new Log(ipfs, testIdentity, { access: testACL, logId: 'X' })
+
+        for (let i = 0; i <= 100; i++) {
+          await log1.append('entry' + i)
+        }
+      })
+
+      it('returns a Symbol.iterator object', async () => {
+        let it = log1.iterator({
+          gte: 'zdpuApFd5XAPkCTmSx7qWQmQzvtdJPtx2K5p9to6ytCS79bfk',
+          amount: 0
+        })
+
+        assert.strictEqual(typeof it[Symbol.iterator], 'function')
+        assert.deepStrictEqual(it.next(), { value: undefined, done: true })
+      })
+
+      it('iterates with lte and explicit amount', async () => {
+        let amount = 10
+
+        let it = log1.iterator({
+          lte: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        assert.strictEqual([...it].length, 10)
+
+        let i = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+        }
+      })
+
+      it('iterates with lt and explicit amount', async () => {
+        let amount = 10
+
+        let it = log1.iterator({
+          lt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 1
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      it('iterates with gt and explicit amount', async () => {
+        let amount = 5
+
+        let it = log1.iterator({
+          gt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (72 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      it('iterates with gte and explicit amount', async () => {
+        let amount = 12
+
+        let it = log1.iterator({
+          gt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (79 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      /* eslint-disable camelcase */
+      it('iterates with combinations of lt, gt, lte, and gte', async () => {
```

Comment:
This test would benefit from being broken down to multiple `it`s. Perhaps one `it` per iterator call or similar.

Reply:
I can do that, yeah... that would fix the variable naming problem above as well.